maven_iuvs/echelle.py

```diff
@@ -636,22 +825,23 @@ def subtract_darks(light_fits, dark_fits):
         good_frame_inds.append(i)
         medians.append(np.median(median_this_frame))
 
-    # Handle what to do with the darks based on whether the second exists.
-    second_dark_exists = True 
-    if np.isnan(second_dark).all():
-        second_dark_exists = False 
-
-    # Now handle the first dark
-    if np.isnan(first_dark).any():  # First dark is bad 
+    # Mark frames as bad if they are associated with the first dark, either by
+    # BEING the first dark, or being the first light, or being the 1-nth light
+    # if the second dark doesn't exist.
+    if not first_dark_good:
         nan_dark_inds.append(0)
         light_frames_with_nan_dark.append(0)
-    else:  # First dark is good 
-        if not second_dark_exists: 
+
+        # A bad 1st dark can't be used for light frames 1-n_int in early files
```

Comment:
I'm not sure we determined that the first dark issue was so bad that we should throw out all the light frames for which no appropriate dark exists. My read of the matching code in the IUVS IDL pipeline suggests that the first dark is used for all frames when no second frame exists.

Dark determination is in the following nested routines:
- `code/superdriver/driver_code/maven_iuvs_process_l1a_to_l1b.pro`
- `code/level1b/iuvs_to_fits_1b.pro`
- `code/level1b/stack_1b_calibrate.pro`
- `code/level1b/iuvs_estimate_dark.pro`

Probably not worth changing to use the IDL procedure now, but maybe worth opening an issue. How many integrations are loss with this treatment?

Reply:
I think very few files have that problem. I agree with you though, I think we should just use the 1st dark. I didn't code that up yet out of caution. It can be an issue.

Reply:
(all ordinals mentioned in this comment assume zero-indexed)

Wait a second - what the comment line you put this comment on is saying is: if there is no 1st dark frame, AND the 0th dark frame is bad/unusable, then there are no valid dark frames. 

Further down in the code, the 0th dark frame---if it's valid and usable---IS used for the 1st through nth light frames if there was only 1 dark frame taken total.

Reply:
where is the 0th dark used? If all of the integers and comments are zero-based, should we change the comments and variable names to reflect this to avoid confusion, or would that only introduce more confusion?

Reply:
OK, I see it on line 879 below. All good.